### PR TITLE
[SPARK-36780][BUILD] Make `dev/mima` runs on Java 17

### PR DESCRIPTION
**Pull request by RabbidHY**
### What changes were proposed in this pull request?

Java 17 has been officially released. This PR makes `dev/mima` runs on Java 17.

### Why are the changes needed?

To make tests pass on Java 17.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual test.

### CODE REVIEW

**Comment by wangyum** [comment]
ok to test.

**Comment by dongjoon-hyun** [comment]
Thank you, @RabbidHY .

File: dev/mima [hunk 1]
```diff
@@ -41,6 +41,12 @@ else
   JAVA_CMD=java
 fi
 
+# Workaround to make mima runs on Java 17+
+JAVA_VER=$(java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\./s///' | cut -d'.' -f1)
+if [[ JAVA_VER -ge 17 ]]; then
```
Comment (dongjoon-hyun) [inline dev/mima:46]: Could you extend this for Java 11, too? `-ge 11` instead of `-ge 17`?

File: dev/mima [hunk 2]
```diff
@@ -41,8 +41,16 @@ else
...
+
+# Workaround to make mima runs on Java 11+
+JAVA_VER=$(java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\./s///' | cut -d'.' -f1)
+if [[ JAVA_VER -ge 11 ]]; then
+  EXTRA_JAVA_OPTIONS="--add-opens=java.base/java.util.jar=ALL-UNNAMED"
+fi
+
 $JAVA_CMD \
   -Xmx2g \
+  $EXTRA_JAVA_OPTIONS \
```
Comment (dongjoon-hyun) [inline dev/mima:53]: Thank you for update. Let's simplify like this by removing line 44~50 and add directly.
```
$JAVA_CMD \
  -Xmx2g \
  -XX:+IgnoreUnrecognizedVMOptions --add-opens=java.base/java.util.jar=ALL-UNNAME \
  ...
```

**Comment by dongjoon-hyun** [comment]
Thank you for updating again, @RabbidHY . I fixed a typo at the last commit. It was due to my comment.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master/3.2.
Thank you, @RabbidHY .

cc @gengliangwang

**Comment by dongjoon-hyun** [comment]
Welcome to the Apache Spark community, @RabbidHY .
You are added to the Apache Spark contributor group and SPARK-36780 is assigned to you. :)

**Comment by RabbidHY** [comment]
> Welcome to the Apache Spark community, @RabbidHY .
> You are added to the Apache Spark contributor group and [SPARK-36780](https://issues.apache.org/jira/browse/SPARK-36780) is assigned to you. :)

Thank you very much Dongjoon! :D

**Pull request closed without merging**